File: 23.01/B2579.java
import java.io.*;

public class B2579 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        int[] arr = new int[n + 1];
        int[] dp = new int[n + 1];

        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(br.readLine());
        }
        if (n > 0) {
            dp[0] = arr[0];
        }
        if (n > 1) {
            dp[1] = arr[0] + arr[1];
        }
        if (n > 2) {
            dp[2] = Math.max(arr[1], arr[0]) + arr[2];
        }

        for (int i = 3; i < n; i++) {
            int a = arr[i - 1] + dp[i - 3];
            int b = dp[i - 2];
            dp[i] = Math.max(a, b) + arr[i];
        }

        System.out.println(dp[n - 1]);
    }
}

File: 23.09/B2579.py
n = int(input())

arr = []

for _ in range(n):
    arr.append(int(input()))
dp = []

dp.append(arr[0])
if (n > 0):
    dp.append(arr[0] + arr[1])
if (n > 1):
    dp.append(max(arr[0], arr[1])+arr[2])

for i in range(3, n):
    dp.append(max(dp[i-2], dp[i-3]+arr[i-1])+arr[i])

print(max(dp))
